File: mobile/socket/messages/server/SupportedStocksMessage.ts
export const SUPPORTED_STOCKS = 'SUPPORTED_STOCKS'

export interface SupportedStocksMessagePayload {
    id: string;
    exchange:string,
    name:string,
    currency:string,
    quoteType:string,
    price: string,
}

export interface SupportedStocksMessage {
  type: typeof SUPPORTED_STOCKS,
  payload: SupportedStocksMessagePayload[]
}

File: stocks-service/stocksserver/messages/server/SupportedStocksMessage.py
SUPPORTED_STOCKS_MESSAGE = "SUPPORTED_STOCKS"

class StockItem:
    def __init__(self, symbol,shortName,exchange,currency,quoteType,regularMarketPrice):
        self.id, self.name, self.exchange, self.currency,self.quoteType, self.price = symbol,shortName,exchange,currency,quoteType,regularMarketPrice
    
class SupportedStocksMessage:
    def __init__(self, payload):
        self.type, self.payload = SUPPORTED_STOCKS_MESSAGE, payload

